File: bouncing-balls.cpp
#include <iostream>
#include <cmath> 
#include <iomanip>

using namespace std;

int main() {
    double startingHeight, firstBounceHeight;
    int numBounces;

    cin >> startingHeight >> firstBounceHeight >> numBounces;

    double bouncinessIndex = firstBounceHeight / startingHeight;

    double totalDistance = 0.0;
    for (int i = 1; i <= numBounces; i++) {
        totalDistance += pow(2, -bouncinessIndex) * (startingHeight + firstBounceHeight);
    }

    cout << fixed << setprecision(6) << totalDistance << endl;

    return 0;
}

File: bouncing-balls.py
def bouncing_balls():
    start_height = float(input())
    first_bounce_height = float(input())
    bounciness_index = first_bounce_height / start_height
    num_bounces = int(input())

    total_distance = 0
    for _ in range(num_bounces):
        total_distance += (1 + bounciness_index) ** (_ * 2)

    return str(round(total_distance, 4))


print(bouncing_balls())